lib/archive/unpack_unix.go

```diff
@@ -202,6 +202,56 @@ func streamCopy(op trace.Operation, stdin io.WriteCloser, tarStream io.Reader) e
 	}
 }
 
+func DockerUnpack(op trace.Operation, root string, tarStream io.Reader) (int64, error) {
+	fi, err := os.Stat(root)
+	if err != nil {
+		// the target unpack path does not exist. We should not get here.
+		op.Errorf("tar unpack target does not exist: %s", root)
+		return 0, err
+	}
+
+	if !fi.IsDir() {
+		err := fmt.Errorf("unpack root target is not a directory: %s", root)
```

Comment:
Does it even make sense to log messages in this function? Would not it be better to just return a specific error? 
```
return 0, fmt.Errorf("unpack root target is not a directory: %s", root)
```

Reply:
Hmm, probably. I must've had a reason for doing this, but it doesn't jump out at me.